src/utils.cpp

```diff
@@ -482,5 +495,241 @@ nlohmann::json getParsedJson(const std::string& pathToJson)
         throw std::runtime_error("Failed to parse JSON file");
     }
 }
+
+void executePostFailAction(const nlohmann::json& i_parsedConfigJson,
```

Comment:
I see it is postAction fail as tag, but in function it is post fail action , could we either have this or that , so that all are common?